hopsworks-common/src/main/java/io/hops/hopsworks/common/dao/hdfs/inode/InodeFacade.java

```diff
@@ -85,107 +93,35 @@ public InodeFacade() {
    * @return
    */
   public List<Inode> findByParent(Inode parent) {
-    TypedQuery<Inode> query = em.createNamedQuery("Inode.findByParentId",
-            Inode.class);
+    TypedQuery<Inode> query = em.createNamedQuery("Inode.findByParentId", Inode.class);
     query.setParameter("parentId", parent.getId());
     return query.getResultList();
   }
+  
+  public Long countByParentId(Inode parent) {
+    TypedQuery<Long> query = em.createNamedQuery("Inode.countByParentId", Long.class);
+    query.setParameter("parentId", parent.getId());
+    return query.getSingleResult();
+  }
```

Comment:
@ErmiasG, can you tell me how this method works? I can't find `countByParentId` query on `Inode.class`. How come you invoke it on the `Long.class`?